src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2728,6 +2730,9 @@ def aks_update(cmd, client, resource_group_name, name,
                 '--rotation-poll-interval can only be specified when azure-keyvault-secrets-provider is enabled')
         azure_keyvault_secrets_provider_addon_profile.config[CONST_ROTATION_POLL_INTERVAL] = rotation_poll_interval
 
+    if nodepool_labels is not None:
+        instance.node_labels = nodepool_labels
```

Comment:
The reason of the failed test case should be the `instance` here is the `ManagedCluster` object but not an `AgentPool` object. You may change this to something like `instance.agent_pool_profiles[0].node_labels = nodepool_labels`. May consider some other checks.